src/Language/PureScript/Ide/State.hs

```diff
@@ -194,11 +196,11 @@ populateStage2 :: (Ide m, MonadLogger m) => m ()
 populateStage2 = do
   st <- ideStateVar <$> ask
   duration <- liftIO $ do
-    start <- getCurrentTime
+    start <- getTime Monotonic
     atomically (populateStage2STM st)
```

Comment:
Well, you're definitely going to have to make sure whatever is getting evaluated actually is evaluated fully before calling `getTime` a second time. You could do that in `populateStage2STM` probably.

A simpler approach might be to benchmark on the command line, using something like the `bench` tool.